### ISSUE TITLE
Error: creating Security Group: InvalidVpcID.NotFound: The vpc ID does not exist

### ISSUE THREAD

**Issue by devopsidiot** [cloudposse/terraform-aws-msk-apache-kafka-cluster#79]
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[![Slack Community](https://slack.cloudposse.com/badge.svg)](https://slack.cloudposse.com)

## Describe the Bug
When applying the terraform module, the module doesn't find or recognize the VPC ID, even though it's there

## Expected Behavior
Security group is created

## Steps to Reproduce
I've created a company module specifically from it w/ this code:
```main.tf
module "msk-apache-kafka-cluster" {
  source  = "cloudposse/msk-apache-kafka-cluster/aws"
  version = "1.1.1"
  
  name = var.name
  vpc_id = var.vpc_id
  subnet_ids = var.subnet_ids
  kafka_version = var.kafka_version
  associated_security_group_ids = var.associated_security_group_ids
  broker_instance_type = var.broker_instance_type
  broker_per_zone = var.broker_per_zone
}
variables.tf
variable "name" {
  type = string
}

variable "vpc_id" {
  type = string
}

variable "subnet_ids" {
  type = list(string)
}

variable "kafka_version" {
  type = string
}

variable "associated_security_group_ids" {
  type = list(string)
}

variable "broker_instance_type" {
  type = string
}

variable "broker_per_zone" {
  type = number
}```
As I am using terragrunt to invoke the tf-module, my terragrunt.hcl looks thusly:
```terraform {
  source = "git@github.com:myplace/terraform-modules.git//snowplow?ref=snowplow/v1.1.0"
}

dependency "vpc" {
  config_path = "../vpc"
}

inputs = {
    name = "myplace-snowplow-test"
    vpc_id = dependency.vpc.outputs.vpc_id
    subnet_ids = dependency.vpc.outputs.private_subnets
    kafka_version = "2.8.1"
    associated_security_group_ids = [dependency.vpc.outputs.default_sg_id]
    broker_instance_type = "kafka.t3.small"
    broker_per_zone = 1
}```
It has a dependency of the outputs of the vpc module that show up like this:
```azs = tolist([
  "us-east-1a",
  "us-east-1b",
  "us-east-1c",
])
cgw_ids = []
database_subnets = [
  "subnet-05ace04da69c0a5c3",
  "subnet-03f094702e6413a5c",
  "subnet-0e29e3ea07b3161bd",
]
default_sg_id = "sg-019db31e6084d695b"
intra_subnets = [
  "subnet-02fa8b695b63f36be",
  "subnet-068a94b0fcb72c6bf",
  "subnet-0edb9a2c27f57b067",
]
nat_public_ips = tolist([
  "3.231.112.27",
])
private_subnets = [
  "subnet-047d998dd1bb4e300",
  "subnet-02627f60507ea09fb",
  "subnet-00ffed109a79644da",
]
public_subnets = [
  "subnet-0b82cf0a6e280600a",
  "subnet-0512c45da9cac36f2",
  "subnet-0588f61d9b5307245",
]
this_customer_gateway = {}
vpc_cidr = "10.2.0.0/16"
vpc_id = "vpc-0adb2021bba46a1c5"```
When I try to run the snowplow module however, I'm getting the following error:
```Error: creating Security Group (myplace-snowplow-test): InvalidVpcID.NotFound: The vpc ID 'vpc-0adb2021bba46a1c5' does not exist
│ 	status code: 400, request id: 3f7c6a9c-0025-4baa-8345-f44496a95c7f
│
│   with module.msk-apache-kafka-cluster.module.broker_security_group.aws_security_group.default[0],
│   on .terraform/modules/msk-apache-kafka-cluster.broker_security_group/main.tf line 24, in resource "aws_security_group" "default":
│   24: resource "aws_security_group" "default" {```
That vpc exists (per the outputs above), and it's in the console as well.  Even when I hardcode that variable in the terragrunt.hcl, it gives the same error.


## Environment (please complete the following information):

MacOS: Ventura
Terraform Version: 1.0.2

**Comment by ivanov-slk** [comment]
I'm getting the same error using the example in the Readme. Probably I'm missing somethign?

**Comment by Nuru** [comment]
I am marking this as closed by #85. If you can reproduce the problem with the current module, please open a new issue, and verify that the VPC not found error is not due to the module being deployed in a different account or region than the VPC was created in.
